checks/raw/pinned_dependencies_test.go

```diff
@@ -2084,3 +2084,212 @@ func TestCollectGitHubActionsWorkflowPinning(t *testing.T) {
+					},
+				},
+			},
+		},
+		{
+			name:        "unpinned by command and 'locked mode' enabled",
+			filenames:   []string{"./dotnet-locked-mode-enabled.csproj"},
+			expectError: false,
+			stagedDependencies: []*checker.Dependency{
+				{
+					Type:   checker.DependencyUseTypeNugetCommand,
+					Pinned: boolAsPointer(false),
+					Remediation: &finding.Remediation{
+						Text: "pin your dependecies by either using a lockfile (https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#locking-dependencies) or by enabling central package management (https://learn.microsoft.com/en-us/nuget/consume-packages/Central-Package-Management)",
+					},
+				},
+			},
+			outcomeDependencies: []checker.Dependency{
+				{
+					Type:        checker.DependencyUseTypeNugetCommand,
+					Pinned:      boolAsPointer(true),
+					Remediation: nil,
+				},
+			},
+		},
+		{
+			name:        "unpinned by command and 'locked mode' enabled and disabled in different csproj files",
+			filenames:   []string{"./dotnet-locked-mode-enabled.csproj", "./dotnet-locked-mode-disabled.csproj"},
+			expectError: false,
+			stagedDependencies: []*checker.Dependency{
+				{
+					Type:        checker.DependencyUseTypeNugetCommand,
+					Pinned:      boolAsPointer(false),
+					Remediation: &finding.Remediation{Text: "remediate"},
+				},
+			},
+			outcomeDependencies: []checker.Dependency{
+				{
+					Type:        checker.DependencyUseTypeNugetCommand,
+					Pinned:      boolAsPointer(false),
+					Remediation: &finding.Remediation{Text: "remediate: some of your csproj files set the RestoreLockedMode property to true, while other do not set it: ./dotnet-locked-mode-disabled.csproj"},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctrl := gomock.NewController(t)
+			mockRepoClient := mockrepo.NewMockRepoClient(ctrl)
+			mockRepoClient.EXPECT().ListFiles(gomock.Any()).Return(tt.filenames, nil).AnyTimes()
+			mockRepoClient.EXPECT().GetDefaultBranchName().Return("main", nil).AnyTimes()
+			mockRepoClient.EXPECT().URI().Return("github.com/ossf/scorecard").AnyTimes()
+			mockRepoClient.EXPECT().GetFileReader(gomock.Any()).AnyTimes().DoAndReturn(func(file string) (io.ReadCloser, error) {
+				return os.Open(filepath.Join("testdata", file))
+			})
+
+			req := checker.CheckRequest{
+				RepoClient: mockRepoClient,
+			}
+
+			err := processCsprojLockedMode(&req, tt.stagedDependencies)
+			if err != nil {
+				if !tt.expectError {
+					t.Error(err.Error())
+				}
+			}
+			t.Log(tt.stagedDependencies)
+			for i := range tt.outcomeDependencies {
+				outcomeDependency := &tt.outcomeDependencies[i]
+				depend := tt.stagedDependencies[i]
+				if diff := cmp.Diff(outcomeDependency, depend); diff != "" {
```

Comment:
note: `cmp.Diff` should be able to handle slice comparison, so you don't need to compare them 1 by 1.